app/controllers/posts_controller.rb

```diff
@@ -1,5 +1,6 @@
 class PostsController < ApplicationController
   before_action :authenticate_user!, except: :index
+  before_action :obtain_post, only: %i[show]
```

Comment:
do we need it if we use it only in one action? but maybe in future perspective let it be